lib/user-interface/react/src/components/chatbot/Chat.tsx

```diff
@@ -295,6 +295,16 @@ export default function Chat({ sessionId }) {
       stop: modelConfig?.stop,
       modelKwargs: modelConfig?.modelKwargs,
     });
+    // LangChain's JS OpenAI client sets default values that we do not want if we didn't set them.
+    // The following parameters are preferred to be null instead of having a numeric default value
+    // as they degrade the quality of the model response if set incorrectly by default.
+    if (!modelConfig?.frequency_penalty && modelConfig?.frequency_penalty !== 0) {
```

Comment:
I sent you a patch file but what I would suggest doing is pulling the properties that we want to null out by defualt out into the modelKwargs property of the modelConfig. These properties will be defaulted to null when passed into modelKwargs and you will get the behavior you are replicating here in a much cleaner way.

Reply:
Awesome suggestion- doing that for the next set of changes. I left the number of tokens alone because those are usually required at the client level, but this is a much nicer approach and we don't need any other custom logic.